### PR TITLE
Latent transformation draft

### PR DESCRIPTION
**Pull request by ajlee21**
This PR includes scripts and data files and output files for analysis to compare linear transformation and latent space transformation:

scripts/
1.  Generate training/test datasets:  generate_input.py
2.  Train VAE and save encoding and decoding models:  tybalt_1layer_model.py
3.  Encode/decode dataset:  apply_saved_model.py
4.  Generate latent space offset:  latent_offset.py
5.  Apply linear/linear offset:  apply_offset.py
6.  Evaluate transformations:  error_geneExp_reconstruction.py

metadata/analysis_steps.txt contains steps of analysis

### CODE REVIEW

File: scripts/apply_saved_model.ipynb [hunk 1]
```diff
@@ -118,11 +754,8 @@
    ],
    "source": [
     "# read in data\n",
```
Comment (huqiwen0313) [inline scripts/apply_saved_model.ipynb]: Brief describe the dataset a little ?

File: scripts/nbconverted/tybalt_1layer_model.py [hunk 2]
```diff
@@ -283,3 +289,27 @@ def on_epoch_end(self, epoch, logs={}):
...
 encoded_rnaseq_df.to_csv(encoded_file, sep='\t')
 
+# Save models
+# (source) https://machinelearningmastery.com/save-load-keras-deep-learning-models/
+
+# Save encoder model
+model_encoder_json = encoder.to_json()
+with open(model_encoder_file, "w") as json_file:
+    json_file.write(model_encoder_json)
+
```
Comment (huqiwen0313) [inline scripts/nbconverted/tybalt_1layer_model.py]: Up to you, I think maybe can also save to hdf5 file
encoder_model_file = path_and_filename
encoder.save(encoder_model_file)

File: scripts/nbconverted/generate_input.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+# In[1]:
+
+
+#-------------------------------------------------------------------------------------------------------------------------------
+# By Alexandra Lee (July 2018) 
+#
+# Generate input datasets
+# Use map_file to group samples into phenotype groups (condition A and B) based on experimental design annotations
+# Then group samples into training and test sets
+#
```
Comment (huqiwen0313) [inline scripts/nbconverted/generate_input.py]: Maybe add a little description of the dataset and phenotype groups

File: scripts/nbconverted/apply_offset.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+# coding: utf-8
+
+# In[1]:
+
+
+#-------------------------------------------------------------------------------------------------------------------------------
+# By Alexandra Lee (July 2018) 
+#
+# Apply offset vector
+#-------------------------------------------------------------------------------------------------------------------------------
```
Comment (huqiwen0313) [inline scripts/nbconverted/apply_offset.py]: Maybe add some descriptions of how to use the script and what is the input/output of the script. Same comments to the other python scripts

File: scripts/nbconverted/error_geneExp_reconstruction.py [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+# In[5]:
+
+
+# Average gene expression across samples
+estimated_data_mean = estimated_data.mean(axis=0)
+obsv_data_mean = obsv_data.mean(axis=0)
+
+#estimated_data_mean_df = pd.DataFrame(estimated_data_mean, index = estimated_data_mean.index, columns=['estimated'])
+#obsv_data_mean_df = pd.DataFrame(obsv_data_mean, index = obsv_data_mean.index, columns = ['observed'])
+
```
Comment (huqiwen0313) [inline scripts/nbconverted/error_geneExp_reconstruction.py]: Can you clean up the codes here a little and delete these lines ?

File: scripts/nbconverted/error_geneExp_reconstruction.py [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+# Try using single sample
+estimated_data_mean_df = pd.DataFrame(estimated_data.iloc[0].transpose())
+obsv_data_mean_df = pd.DataFrame(obsv_data.iloc[0].transpose())
+estimated_data_mean_df.columns = ['estimated']
+obsv_data_mean_df.columns = ['observed']
+
+estimated_data_mean_df.head(5)
+#obsv_data_mean_df.head(5)
+
```
Comment (huqiwen0313) [inline scripts/nbconverted/error_geneExp_reconstruction.py]: also here

File: scripts/nbconverted/error_geneExp_reconstruction.py [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+X.head(5)
+
+
+# In[7]:
+
+
+# Plot
+fg=sns.jointplot(x='estimated', y='observed', data=X, kind='reg');
+
+#fg = sns.lmplot(x = 'estimated', y = 'observed', data = X)
```
Comment (huqiwen0313) [inline scripts/nbconverted/error_geneExp_reconstruction.py]: here too, same comments for the other parts

**Review by huqiwen0313 — COMMENTED** [review]
Great PR ! Only some minor comments

File: scripts/nbconverted/error_geneExp_reconstruction.py (hunk 6 above)
Comment (ajlee21) [reply scripts/nbconverted/error_geneExp_reconstruction.py]: This is for when I was spot checking what the data looks like

File: scripts/nbconverted/tybalt_1layer_model.py (hunk 2 above)
Comment (ajlee21) [reply scripts/nbconverted/tybalt_1layer_model.py]: I was just following the example for this.  I am not sure what the difference is in the file types...do you have a suggestion?

**Comment by ajlee21** [comment]
Modified the scripts based on your comments and had a couple of questions for you

File: scripts/nbconverted/tybalt_1layer_model.py (hunk 2 above)
Comment (huqiwen0313) [reply scripts/nbconverted/tybalt_1layer_model.py]: I know hdf5 format is used to save large data (genomic data).. .jason is a format that can be used to store different structures. I think hdf5 maybe more suitable for storing large data, but I am that sure. It is up to you.

**Review by huqiwen0313 — APPROVED** [review]
Modification looks good.

**Comment by ajlee21** [comment]
Thanks Qiwen,  I modified the script to use the hdf5 format

**Pull request merged**